internal/plugin/grpc_provider.go

```diff
@@ -1251,8 +1253,94 @@ func (p *GRPCProvider) CallFunction(r providers.CallFunctionRequest) (resp provi
 	return resp
 }
 
-func (p *GRPCProvider) ListResource(r providers.ListResourceRequest) error {
-	panic("not implemented")
+func (p *GRPCProvider) ListResource(r providers.ListResourceRequest) (providers.ListResourceResponse, error) {
+	logger.Trace("GRPCProvider: ListResource")
+
+	schema := p.GetProviderSchema()
+	if schema.Diagnostics.HasErrors() {
+		return nil, schema.Diagnostics.Err()
+	}
+
+	listResourceSchema, ok := schema.ListResourceTypes[r.TypeName]
+	if !ok {
+		return nil, fmt.Errorf("unknown list resource type %q", r.TypeName)
+	}
+
+	resourceSchema, ok := schema.ResourceTypes[r.TypeName]
+	if !ok || resourceSchema.Identity == nil {
+		return nil, fmt.Errorf("identity schema not found for resource type %s", r.TypeName)
+	}
+
+	mp, err := msgpack.Marshal(r.Config, listResourceSchema.Body.ImpliedType())
+	if err != nil {
+		return nil, err
+	}
+
+	protoReq := &proto.ListResource_Request{
+		TypeName:              r.TypeName,
+		Config:                &proto.DynamicValue{Msgpack: mp},
+		IncludeResourceObject: r.IncludeResourceObject,
+	}
+
+	// Start the streaming RPC
+	client, err := p.client.ListResource(p.ctx, protoReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(yield func(providers.ListResourceEvent) bool) {
```

Comment:
(somewhat repeated from the parallel PR) while I definitely love the use of the iterator, I think it might be more convenient in core to hide this implementation detail. Core is always going to want to operate on the result as a single object, and the stream is only there because of single call size limitations, not because core actually needs a stream of results.

Reply:
> Core is always going to want to operate on the result as a single object

is the ability to stream data back to the renderer more complicated than we assumed? Just wondering what would be the limitation to support that